library/cisco_imc_ipmi.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: cisco_imc_ipmi
+short_description: Configures ipmi a Cisco IMC Server
+version_added: "0.9.0.0"
+description:
+    - Configures the Serial Over Lan(SOL) service on a Cisco IMC Server
+Input Params:
+    priv:
+        description: Privilege to be used
+        required: False
+        choices: ["admin", "user", "read-only"]
+        default: 'user'
+
+    key:
+        description: Hexadecimal Key to be used for authentication
+        required: False
+
+    server_id:
+        description: Server Id to be specified for C3260 platforms
+        choices: ["admin", "read-only", "user"]
+        default: "read-only"
+        required: False
+
+    state:
+        description: Used to create or delete the SOL console
+        choices: ["present", "absent"]
+        default: "present"
+        required: False
+
+requirements: ['imcsdk']
+author: "Branson Matheson (brmathes@cisco.com)"
+'''
+
+EXAMPLES = '''
+- name: enable IPMI
+  cisco_imc_ipmi:
+    priv: "admin"
+    state: "present"
+    ip: "192.168.1.1"
+    username: "admin"
+    password: "password"
+'''
+
+
+def login(module):
```

Comment:
this was copied directly from cisco_imc_ntp. Probably should be cleaned up as well